backend/pkg/modules/internal/traceanalyzer/nlid/nlid.go

```diff
@@ -0,0 +1,280 @@
+	// - Body, XXX Don't do it right now
+
+	r := ph
+	r.Do(func(p interface{}) {
+		if p == nil {
+			return
+		}
+		prevParams, ok := p.(params)
+		if !ok {
+			return
+		}
+
+		for param := range reqParams {
+			if prevParams[param] {
+				// This parameter was already present, that OK, it's not an NLID
+				// Remove it from the parameters to checks
+				reqParams[param] = false // Mark this parameter as already learnt
+				// delete(reqParams, kReq)
+			}
+		}
+	})
+
+	// Here, if reqParams is empty, this means that all parameters from the
+	// Request were found in the history, there is not observation to
+	// return. The parameters that are left in reqParams are the one which
+	// were not found in the history, meaning that they are non learnt IDs
+	for k := range reqParams {
+		if reqParams[k] {
+			eventAnns = append(eventAnns, core.Annotation{
+				Name:       "NLID",
+				Annotation: []byte(k),
+			})
+		}
+	}
+
+	return eventAnns
+}
+
+// We store an id if:
+// - It's in the Response headers AND it looks like an ID.
+// - It's in the Query parameters AND it looks like an ID.
+// - It's in the flattened Response Body AND it's a number or a string that looks like an ID.
+func (n *NLID) learnIDs(trace pluginsmodels.Telemetry) {
+	params := make(params)
+	api := getAPI(trace)
+
+	// Get all parameters of the response
+	// - Header
+	for _, h := range trace.Response.Common.Headers {
+		if maybeID(h.Key, h.Value) {
+			params[h.Value] = true
+		}
+	}
+
+	// Query Parameters
+	u, err := url.Parse(trace.Request.Path)
+	if err == nil {
+		m, _ := url.ParseQuery(u.RawQuery)
+		for k, v := range m {
+			if maybeID(k, v[0]) {
+				params[v[0]] = true // XXX: only get the first value of each query parameter
+			}
+		}
+	}
+	// - Learn Response Body parameters
+	if !trace.Response.Common.TruncatedBody && len(trace.Response.Common.Body) > 0 {
+		if err = getBodyParams(&params, trace.Response.Common.Body); nil != err {
+			// Log the problem, but continue anyway, it's not blocking.
+			log.Debugf("unable to get parameters from body: %v", err)
+		}
+	}
+
+	if _, found := n.paramsHistory[api]; !found {
+		n.paramsHistory[api] = ring.New(n.historySize)
+	}
+	n.paramsHistory[api].Value = params
+	n.paramsHistory[api] = n.paramsHistory[api].Next()
+}
+
+func (n *NLID) skipTrace(trace *pluginsmodels.Telemetry) bool {
+	return false
+}
+
+func getBodyParams(params *params, body []byte) error {
+	var parsed interface{}
+
+	// We deserialize the JSON object this way because json.Unmarshal doesn't
+	// distinguish between int and floats. Here, thanks to d.UseNumber(), we
+	// can switch on json.Number and then try to cast to Int64.
+	d := json.NewDecoder(bytes.NewReader(body))
+	d.UseNumber()
+	if err := d.Decode(&parsed); err != nil {
+		return fmt.Errorf("unable to decode body: %w", err)
+	}
+
+	getDecodedBodyParams(params, parsed, "")
+
+	return nil
+}
+
+func getDecodedBodyParams(o *map[string]bool, val interface{}, prefix string) {
```

Comment:
```suggestion
func getDecodedBodyParams(o map[string]bool, val interface{}, prefix string) {
```
No need for passing map as a pointer since map is already a reference